purple-pill/js/controllers/fontexportController.js

```diff
@@ -32,140 +112,107 @@ function($scope, $http, sharedScope, $timeout) {
-              ;
-            io[index] = exportObjects[1];
-            return gen;
-        }
-
-        var current_instance = 0
-          , total_instances = instances_for_export.length
-          , glyphs_phase_percentage = 80 //An estimated 80% of the time is taken generating the files. The other 20% is spent zipping the results.
-          , UI_UPDATE_TIMESLICE = 50 // msecs
-          ;
-          
-        function setProgress(width, text) {
-            $("#progressbar").animate({"opacity": 1, "width": width + "%"}, /*duration:*/ UI_UPDATE_TIMESLICE);
-            if (text)
-                $("#progresslabel").html(text);
-        }
-
         function setDownloadBlobLink(text, blob, filename) {
-            $("#progressbar").animate({"width": "100%", "opacity": 1}, /*duration:*/ UI_UPDATE_TIMESLICE);
-            $("#progresslabel").html("");
-            $("#blob_download").css("display", "block");
-            $("#blob_download").children("a").html(text).click(function(){
+            var download = $("#blob_download");
+            progress.complete();
+
+            download.css("display", "block");
+            download.children("a").html(text).click(function(){
                 $scope.data.stateless.saveAs(blob, filename);
-                resetProgressBar();
-                delete bundle_data;
+                progress.reset();
+                download.css("display", "none").children("a").unbind("click");
+                delete bundleData;
             });
         }
-        
-        function resetProgressBar() {
-            $("#progressbar").animate({"opacity": 0, "width": 0}, /*duration:*/ 0); // (that means "do it immediately!")
-            $("#progresslabel").html("");
-            $("#blob_download").css("display", "none").children("a").unbind("click");
+
+        function exportingGlyphMessage (it, instanceIndex){
+            var msg
+              , currentGlyph = it.value['current_glyph'] + 1 //humans start counting from 1.
+              , totalGlyphs = it.value['total_glyphs']
+              , glyphId = it.value['glyph_id']
+              ;
+            msg = totalInstances + " instances to export."
+            msg += "Calculating glyph '" + glyphId + "' (" + currentGlyph + " of " + totalGlyphs + ")"
+            msg += " of instance #" + (instanceIndex+1);
+            return msg;
+        }
+
+        function calculateGlyphsProgress(it, instanceIndex, totalInstances){
+            var percentage
+              , currentGlyph = it.value['current_glyph']
+              , totalGlyphs = it.value['total_glyphs']
+              ;
+            percentage = (instanceIndex + (currentGlyph / totalGlyphs)) / totalInstances;
+            return percentage;
         }
 
-        function exportFont_compute_glyphs(){
+        var index = 0
+          , totalInstances = exportObjects.length
+          , obj = exportObjects.pop()
+          ;
+        function exportFontComputeGlyphs(){
```

Comment:
Add also the arguments `bundle` and `callback`(instead of callback, you could also figure out how to do this using a promise, but I think passing a callback down is ok here.)